lib/rspec/determinator.rb

```diff
@@ -26,54 +35,45 @@ def forced_determination(name, outcome, only_for: {})
           outcome = send(outcome) if outcome.is_a?(Symbol)
           only_for = send(only_for) if only_for.is_a?(Symbol)
 
-          fake_determinator.mock_result(
-            name,
-            outcome,
-            only_for: only_for
-          )
+          ::RSpec::Determinator::FakeDeterminator.new(fake_retriever).mock_result(name, outcome, only_for: only_for)
         end
       end
-    end
-
-    class FakeControl
-      def initialize
-        @mocked_results = Hash.new { |h, k| h[k] = {} }
-      end
-
-      def for_actor(**args)
-        ::Determinator::ActorControl.new(self, **args)
-      end
-
-      def mock_result(name, result, only_for: {})
-        @mocked_results[name.to_s][normalize_properties(only_for)] = result
-      end
 
-      def fake_determinate(name, id: nil, guid: nil, properties: {})
-        properties[:id] = id if id
-        properties[:guid] = guid if guid
+    end
 
-        outcome_for_feature_given_properties(name.to_s, normalize_properties(properties))
+    class FakeDeterminator
+      def initialize(in_memory_retriever)
+        @retriever = in_memory_retriever
       end
-      alias_method :feature_flag_on?, :fake_determinate
-      alias_method :which_variant, :fake_determinate
 
-      private
+      def mock_result(name, outcome, only_for: {})
+        active = case outcome
+                 when true, false then
+                   outcome
+                 else
+                   true
+                 end
+        variants = case outcome
+                   when true, false then
+                     [ ]
+                   else
+                     { outcome => 1 }
+                   end
+        target_group = ::Determinator::TargetGroup.new(
+          rollout: 65_536,
+          constraints: Hash[ only_for.map { |key, value| [ key.to_s, value ] } ]
```

Comment:
`normalize_properties` converted the values to strings, where this doesn't - I bet that's why tests are failing when using only for

Reply:
Fixed this - thanks!